### ISSUE TITLE
Error during 'npm install'

### ISSUE THREAD

**Issue by willbprog127** [plan8studios/encrypted-chat#1]
Thanks for this nice project.  The demo looks like just what I need, except when I try to build per your instructions, I receive this error when running `npm install`:

```
npm ERR! Linux 4.4.0-141-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! node v6.16.0
npm ERR! npm  v3.10.10
npm ERR! code E404

npm ERR! 404 Not found : @fortawesome/pro-regular-svg-icons
npm ERR! 404 
npm ERR! 404  '@fortawesome/pro-regular-svg-icons' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 It was specified as a dependency of 'encrypted-chat'
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! Please include the following file with any support request:
npm ERR!     /srv/[private]/echat/npm-debug.log
```
I am *not* a node.js expert so I don't know what to do next...

Also, the `npm-debug.log` does not exist, otherwise I'd include it.

Thank you! 👍

**Comment by geoffoliver** [comment]
Ah! Good catch. That's because I'm using the FontAwesome pro icons. You've got two options for how to fix this:
- Buy FontAwesome pro (not ideal)
- Swap out the pro icons for free set (free and pretty easy)

Feel free to make the change yourself and send a pull request! Otherwise, if you can wait, I can probably get this fixed up this weekend.

**Comment by willbprog127** [comment]
> * Swap out the pro icons for free set (free and pretty easy)

Thanks for your reply! 😄 

I'd be happy to do this if I knew what to do.  As noted earlier, I'm not very well-versed in things node, so I wouldn't be able to create a pull-request.

I did search for other icons and tried substituting them in `package.json` but then when I run `npm install` it runs for a while then dies with `Killed` with no error.

**Comment by geoffoliver** [comment]
Yeah, it’s a bit more involved than that :-) I’ll take care of this over the weekend. Thanks again for finding the bug!

**Comment by geoffoliver** [comment]
This should be resolved with [this commit](https://github.com/plan8studios/encrypted-chat/commit/0271c8a5c5c49ef3fecea1d7daa2cde0fc205e85)

**Comment by willbprog127** [comment]
Thank you!  I'll check it out and see how it goes.  Thanks again! :+1:

**Comment by willbprog127** [comment]
I've checked it out this evening and it's giving a different error.  If you need me to open a separate issue, I can do that.  The error is:

```
$ npm install
npm ERR! prepareGitDep 1> 
npm ERR! prepareGitDep > asmcrypto.js@2.3.0 prepare /home/ee/.npm/_cacache/tmp/git-clone-e544762a
npm ERR! prepareGitDep > node -r esm build.js
npm ERR! prepareGitDep 
npm ERR! prepareGitDep 
npm ERR! prepareGitDep 2> npm WARN install Usage of the `--dev` option is deprecated. Use `--only=dev` instead.
npm ERR! prepareGitDep Aborted (core dumped)
npm ERR! prepareGitDep npm ERR! code ELIFECYCLE
npm ERR! prepareGitDep npm ERR! errno 134
npm ERR! prepareGitDep npm ERR! asmcrypto.js@2.3.0 prepare: `node -r esm build.js`
npm ERR! prepareGitDep npm ERR! Exit status 134
npm ERR! prepareGitDep npm ERR! 
npm ERR! prepareGitDep npm ERR! Failed at the asmcrypto.js@2.3.0 prepare script.
npm ERR! prepareGitDep npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! prepareGitDep 
npm ERR! prepareGitDep npm ERR! A complete log of this run can be found in:
npm ERR! prepareGitDep npm ERR!     /home/ee/.npm/_logs/2019-02-15T07_03_02_224Z-debug.log
npm ERR! prepareGitDep 
npm ERR! code ENOPACKAGEJSON
npm ERR! package.json Non-registry package missing package.json: asmcrypto.js@github:openpgpjs/asmcrypto#6e4e407b9b8ae317925a9e677cc7b4de3e447e83.
npm ERR! package.json npm can't find a package.json file in your current directory.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ee/.npm/_logs/2019-02-15T07_03_03_572Z-debug.log
```
System info:
-
* Ubuntu 18.04.2 LTS (Bionic Beaver) 32-bit VM
* npm version 6.4.1
* node.js version v8.15.0 (nodesource)

[2019-02-15T07_03_03_572Z-debug.log](https://github.com/plan8studios/encrypted-chat/files/2868134/2019-02-15T07_03_03_572Z-debug.log)

**Comment by geoffoliver** [comment]
Well that’s weird. A new issue would be great :-) I’ll dig into this ASAP! Thanks for being my beta tester.

**Comment by willbprog127** [comment]
> Thanks for being my beta tester.

My pleasure.  :+1:   I understand because I do software development, but on the other side of the language spectrum.

**Comment by willbprog127** [comment]
After finding that part of issue #2 was memory exhaustion in my very small VM instance, I performed `npm install` without issue.  However, when I run `npm run build` I get the following error:
```
ERROR in src/app/app.module.ts(9,88): error TS2307: Cannot find module '@fortawesome/pro-solid-svg-icons'
```

**Comment by geoffoliver** [comment]
I just pushed a fix. Give that a shot :)

**Comment by willbprog127** [comment]
All good!  It built properly on 64-bit! :tada: 

Now to spin up a 32-bit VM, build it there and then copy to my production VM and see how it goes.

Thank you so much! :smile:

**Comment by geoffoliver** [comment]
🤞Good luck!!

**Comment by willbprog127** [comment]
When I first tried to run the server it required me to manually install these packages.  Unsure if they were supposed to be installed with `npm install` or ...?
* node-static
* socket.io
* md5
* uuid/v4

After installing those packages, I finally was able to get Encrypted Chat working, and on my main workstation (Firefox and Gentoo Linux) I had no problem loading the Chat page.  I then tried to chat with myself using an Ubuntu 18.04 VM and it gave me invalid issuer errors for my very valid Let's Encrypt cert that I use for other services.  I then tried from a remote computer in another state and it also gave the 'invalid issuer' error when trying to view Encrypted Chat on Firefox, but on yet another computer on that same remote network Encrypted Chat loaded and worked just fine.  I'm unsure what the deal is with the cert situation as it works fine for the rest of our site.

All that aside, it is working fine now on most computers and my iPhone.  I haven't had a chance to test it on Mac or Windows yet but will try this weekend.

Thanks so much for this project! :+1:
